File: 8425/bai123.py
def latsach(a, b):
    t=a//2
    l=b//2
    r = t - l
    return min(l,r)
with open("bai1.inp", "r") as fi, open("bai1.out", "w") as fo:
    n, m = map(int, fi.readline().split())
    fo.write(latsach(n, m))
    fo.write("\n")

File: 10425/bai123.cpp
#include <bits/stdc++.h>
using namespace std;

int latsach(int n, int p) {
    return min(p / 2, n / 2 - p / 2);
}


int main() {
    freopen("bai1.inp", "r", stdin);
    freopen("bai1.out", "w", stdout);

    int n, m;
    cin >> n >> m;

    cout << latsach(n, m) << endl;

    return 0;
}
